internal/engine/service_watch.go

```diff
@@ -45,7 +46,9 @@ func (w *ServiceWatcher) OnChange(ctx context.Context, st store.RStore) {
 	}
 	w.watching = true
 
-	ch, err := w.kCli.WatchServices(ctx, []model.LabelPair{k8s.TiltRunLabel()})
+	ctx2, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+	ch, err := w.kCli.WatchServices(ctx2, []model.LabelPair{k8s.TiltRunLabel()})
 	if err != nil {
 		err = errors.Wrap(err, "Error watching services. Are you connected to kubernetes?\n")
```

Comment:
"Are you connected to Kubernetes" is not a helpful error message if we timed out, I might remove that bit and trust the `err` to contain the relevant context.

Reply:
fwiw, i thought we added that precisely because we got user reports that they didn't understand the error message. there are a wide variety of weird error messages you can get from this function

Reply:
Right, but if we get this error due to a timeout, it doesn't imply that we're not connected to k8s, right, but rather that something ELSE is wrong? in that case, the hint "are you connected to k8s" sends the user down a rabbit hole.